public/js/selfoss-ui.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * ui change functions
+ */
+selfoss.ui = {
+
+
+    entryStarr: function(id, starred) {
+        var button = $("#entry"+id+" .entry-starr, #entrr"+id+" .entry-starr");
+
+        // update button
+        if(starred) {
+            button.addClass('active');
+            button.html($('#lang').data('unstar'));
+        } else {
+            button.removeClass('active');
+            button.html($('#lang').data('star'));
+        }
+    },
+
+
+    entryMark: function(id, unread) {
+        var button = $("#entry"+id+" .entry-unread, #entrr"+id+" .entry-unread");
+        var parent = $("#entry"+id+", #entrr"+id);
+
+        // update button and entry style
+        if(unread) {
+            button.addClass('active');
+            button.html($('#lang').data('mark'));
+            parent.addClass('unread');
+        } else {
+            button.removeClass('active');
+            button.html($('#lang').data('unmark'));
+            parent.removeClass('unread');
+        }
+    },
+
+
+    refreshItemStatuses: function(entry_statuses) {
+        $('.entry').each(function(index, item) {
+            var id = $(this).data(('entry-id'));
+            new_status = entry_statuses.find(function(entry_status) {
+                return entry_status.id == id;
+            });
+            if( new_status ) {
+                selfoss.ui.entryStarr(id, new_status.starr);
+                selfoss.ui.entryMark(id, new_status.unread);
+            }
+        });
+    },
+
+
+    refreshStreamButtons: function(entries=false,
```

Comment:
Nice, I did not know that any browsers already support this syntax. [IE does not](https://kangax.github.io/compat-table/es6/#test-default_function_parameters), though so we’d better stay at ES5 for now.